### PR TITLE
Native platform support using socket communication

### PR DESCRIPTION
**Pull request by nicolas-rabault**
# PR Checklist 

### _1- PR Comments_
    *If needed, to be filled by developer*  

<br><hr/>
*WARNING : Do not edit the checklist below.*  
<hr/><br>

### _2- Developer section_

- [x] [Documentation] is up to date with new feature
- [x] [Tests] are *passed OK* (Non regression, new features & bug fixes)
- [x] [Code Quality] Remind to check if :
    * Each function has a header (description, inputs, outputs) 
    * Code is commented (particularly in hard to understand areas)
    * There are no new warnings that can be corrected
    * Commits policy is respected (constitancy commits, clear commits comments)
<hr/><br>

### _3- QA section_

- [x] [Review] tests for new features have been *reviewed*
- [x] [Changelog] is up-to-date with expected tags  
    🆕 Feature: [Feature] Description...  
    🆕 Added: [Feature] Description...  
    🆕 Changed: [Feature] Description...  
    🛠️ Fix: [Feature] Description...

### CODE REVIEW

File: network/robus/src/msg_alloc.c [hunk 1]
```diff
@@ -44,6 +44,7 @@
 
 #include <string.h>
 #include <stdbool.h>
+#include <limits.h>
```
Comment (Salem-Luos) [inline network/robus/src/msg_alloc.c:47]: Why this headers is needed?

File: engine/HAL/NATIVE/luos_hal_config.h [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ *        Do not modify this file if you want to ovewrite change define in you project
+ * @Family x86
+ * @author Luos
+ * @version 0.0.0
+ ******************************************************************************/
+#ifndef _LUOSHAL_CONFIG_H_
+#define _LUOSHAL_CONFIG_H_
+
+#ifndef MCUFREQ
+#define MCUFREQ 100000000 // MCU frequence
```
Comment (Salem-Luos) [inline engine/HAL/NATIVE/luos_hal_config.h:14]: not really important but how you define this?

File: engine/HAL/NATIVE/luos_hal_config.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+#define ADDRESS_LAST_PAGE_FLASH (uint32_t) last_page_stub_flash_x86
+#endif
+
+/*******************************************************************************
+ * BOOTLOADER CONFIG
+ ******************************************************************************/
+#define FLASH_END FLASH_SIZE - 1
+
+#ifndef END_ERASE_BOOTLOADER
+#define END_ERASE_BOOTLOADER (uint32_t)0x08020000
```
Comment (Salem-Luos) [inline engine/HAL/NATIVE/luos_hal_config.h:45]: same for address? if not needed maybe we should remove them

File: network/robus/src/msg_alloc.c (hunk 1 above)
Comment (nicolas-rabault) [reply network/robus/src/msg_alloc.c:47]: Because we need it to get `INT_MAX`

File: engine/HAL/NATIVE/luos_hal_config.h (hunk 2 above)
Comment (nicolas-rabault) [reply engine/HAL/NATIVE/luos_hal_config.h:14]: I any case I will remake a lot of things on those files.
This PR is far from ready....

**Review by JeromeGalan — APPROVED** [review]

**Pull request merged**